WordPress/src/main/java/org/wordpress/android/datasets/CommentTable.java

```diff
@@ -150,6 +151,50 @@ public static CommentList getCommentsForBlog(int localBlogId) {
     }
 
     /**
+     * get comments for a blog that have a specific status
+     * @param localBlogId - unique id in account table for this blog
+     * @param filter - status to filter comments by
+     * @return list of comments for this blog
+     */
+    public static CommentList getCommentsForBlogWithFilter(int localBlogId, CommentStatus filter) {
+        CommentList comments = new CommentList();
+        Cursor c;
+
+        //aggregating 'all' to include approved and unapproved comments
+        if (CommentStatus.UNKNOWN.equals(filter)){
+            //we need to get the filter values for both XMLrpc and REST api as in the case of a migration where existing
+            // data is present on a device, we still need to be able to filter both values
+            String[] args = {Integer.toString(localBlogId),
+                    CommentStatus.toString(CommentStatus.APPROVED),
+                    CommentStatus.toString(CommentStatus.UNAPPROVED),
+                    CommentStatus.toRESTString(CommentStatus.APPROVED),
+                    CommentStatus.toRESTString(CommentStatus.UNAPPROVED)};
+            c = getReadableDb().rawQuery(
+                    "SELECT * FROM " + COMMENTS_TABLE + " WHERE blog_id=? AND (status=? OR status=? OR status=? OR status=?)  ORDER BY published DESC", args);
+
+        } else {
+            //we need to get the filter values for both XMLrpc and REST api as in the case of a migration where existing
+            // data is present on a device, we still need to be able to filter both values
+            String[] args = {Integer.toString(localBlogId), CommentStatus.toString(filter), CommentStatus.toRESTString(filter)};
+            c = getReadableDb().rawQuery(
+                    "SELECT * FROM " + COMMENTS_TABLE + " WHERE blog_id=? AND (status=? OR status=?)  ORDER BY published DESC", args);
+        }
+
+        try {
+            if (c.moveToFirst()) {
```

Comment:
Minor nit: since `moveToNext` will move to the first row on a new cursor, this can be simplified to:
```
while (c.moveToNext()) {
     Comment comment = getCommentFromCursor(c);
     comments.add(comment);
}
```